### PR TITLE
sessionctx/variable: use sysvar api correctly for tidb_row_format_version

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/35051

Problem Summary:

This fixes incorrect usage of the sysvar API for the variable `tidb_row_format_version`. It is left over from a historical refactor, and along with https://github.com/pingcap/tidb/pull/35052 fixes it so that global-only variables are not loaded into the session.

### What is changed and how it works?

The global-only sysvar `tidb_row_format_version` is improved in usage.

~The behavior is slightly different, since if a user changed the value of `tidb_row_format_version` it would change `vars.RowEncoder.Enable` for that session. Instead, new sesions are now changed. This doesn't seem to have any impact on existing tests, but needs someone who works on this code to confirm as well.~

**Nevermind:** because it's SetSession, but the scope is SetGlobal, it's the same behavior. The SetSession is called on init for new sessions. However, it does behave slightly differently for lightning, which calls `vars.SetSystemVar()` directly. 

### Check List

Tests 

- [x] Unit test (covered by existing tests)
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
The scope of tidb_row_format_version changes to session or global. This is required because lightning is using it as a session variable currently. See: br/pkg/lightning/backend/kv/sql2kv_test.go:TestEncode() -> NewTableKVEncoder() is called with the sysvar version 1, which in newSession() calls vars.SetSystemVar() to change it on a session basis. This is not expected use of the API, since validation is expected to be called before that function (and would have failed).
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Several global-only system variables were previously loaded into new sessions and cached. They now behave more compatible with MySQL and remain in global scope.
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]
LGTM, but lighting guys deserve a notification.

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1007,6 +997,18 @@ var defaultSysVars = []*SysVar{
 	}},
 	{Scope: ScopeGlobal | ScopeSession, Name: BlockEncryptionMode, Value: "aes-128-ecb"},
 	/* TiDB specific variables */
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBRowFormatVersion, Value: strconv.Itoa(DefTiDBRowFormatV1), Type: TypeUnsigned, MinValue: 1, MaxValue: 2, SetGlobal: func(s *SessionVars, val string) error {
```
Comment (bb7133) [inline sessionctx/variable/sysvar.go:1000]: Could you explain why 'ScopeSession' is introduced? IMHO it might be dangerous to change it in a session.

Comment (morgo) [reply sessionctx/variable/sysvar.go:1000]: It _is_ already a session scoped variable, it just uses the API incorrectly:

- It has special exception to load into the session in `SkipInit()` which I am trying to refactor out here. This is left over from https://github.com/pingcap/tidb/pull/24836
- It has a SetSession function on the struct.
- Lightning calls vars.SetSystemVar() directly which calls the SetSession function, and skips the validation check to confirm that ScopeSession is permitted.

**Comment by morgo** [comment]
Fixed in https://github.com/pingcap/tidb/pull/36640

**Pull request closed without merging**